aya/src/bpf.rs

```diff
@@ -58,7 +60,243 @@ pub(crate) struct bpf_map_def {
     pub(crate) map_flags: u32,
     // optional features
     pub(crate) id: u32,
-    pub(crate) pinning: u32,
+    pub(crate) pinning: BpfPinningType,
+}
+
+#[repr(u32)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub(crate) enum BpfPinningType {
+    PinNone = 0,
+    PinByName = 1,
+}
+
+impl TryFrom<u32> for BpfPinningType {
+    type Error = BpfError;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(BpfPinningType::PinNone),
+            1 => Ok(BpfPinningType::PinByName),
+            n => Err(BpfError::UnexpectedPinningType { name: n }),
+        }
+    }
+}
+
+impl Default for BpfPinningType {
+    fn default() -> Self {
+        BpfPinningType::PinNone
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct BpfLoader {
+    btf: Option<Btf>,
+    map_pin_path: Option<PathBuf>,
+}
+
+impl BpfLoader {
+    pub fn new() -> BpfLoader {
+        BpfLoader {
+            btf: None,
+            map_pin_path: None,
+        }
+    }
+
+    // Set the target BTF
+    pub fn btf(&mut self, btf: Btf) -> &mut BpfLoader {
+        self.btf = Some(btf);
+        self
+    }
+
+    // Set the target BTF to /sys/kernel/btf/vmlinux
+    pub fn default_btf(&mut self) -> &mut BpfLoader {
```

Comment:
I don't think we need this, I think we can let the caller use `Btf::from_sys_fs()` and let them figure out whether they want to error out if BTF is not available or not.